### PR TITLE
Confidence score htr

### PR DESCRIPTION
**Pull request by starride-teklia**
We are interested in the confidence scores associated with HTR predictions. 

To implement this, I updated these files:
* `laia/decoders/ctc_greedy_decoder.py` now returns the character-based probabilities (after removing duplicates and <ctc> tokens)
* `laia/callbacks/decode.py` can now print the overall confidence score (= averaged character probabilities)
* `laia/common/arguments.py` now includes the argument decode.print_confidence_scores
* `laia/scripts/htr/decode_ctc.py` `Decode` is called using the argument decode.print_confidence_scores

I also updated black in `.pre-commit-config.yaml` as I ran into [this issue](https://stackoverflow.com/questions/71673404/importerror-cannot-import-name-unicodefun-from-click).

### CODE REVIEW

File: laia/callbacks/decode.py [hunk 1]
```diff
@@ -1,12 +1,31 @@
...
+import numpy as np
 import pytorch_lightning as pl
 from tqdm.auto import tqdm
 
 from laia.decoders import CTCGreedyDecoder
 from laia.utils import SymbolsTable
 
 
+def compute_word_prob(symbols, hyp, prob, input_separator):
+    # compute mean confidence score and mean confidence score by word
```
Comment (La0) [inline laia/callbacks/decode.py:12]: - Use `"""` docstring style here
- Please describe the output (list of confidence scores)

File: laia/callbacks/decode.py [hunk 2]
```diff
@@ -34,12 +55,27 @@ def __init__(
...
         img_ids = pl_module.batch_id_fn(batch)
         hyps = self.decoder(outputs)["hyp"]
-        for i, (img_id, hyp) in enumerate(zip(img_ids, hyps)):
+        probs = self.decoder(outputs)["prob-htr-char"]
+
+        # compute mean confidence score
+        line_probs = [np.mean(prob) for prob in probs]
+
+        # compute mean confidence score by word
+        word_probs = [
```
Comment (La0) [inline laia/callbacks/decode.py:71]:
```suggestion
        word_probs = [
            compute_word_prob(self.syms, hyp, prob, self.input_space)
            for prob, hyp in zip(probs, hyps)
        ]
```

File: laia/decoders/ctc_greedy_decoder.py [hunk 3]
```diff
@@ -6,23 +6,54 @@
...
             out["segmentation"] = [
                 CTCGreedyDecoder.compute_segmentation(x_n.indices.tolist()) for x_n in x
             ]
+
+        probs = [x_n.values.exp() for x_n in x]
         x = [x_n.indices for x_n in x]
+
         # Remove repeated symbols
-        x = [torch.unique_consecutive(x_n) for x_n in x]
+        zero_tensor = torch.tensor([0]).cuda() if x[0].is_cuda else torch.tensor([0])
```
Comment (La0) [inline laia/decoders/ctc_greedy_decoder.py:34]: The 2 following code blocks are hard to read due to their density. You can simply let the code "breathe" by adding new lines, or making it more explicit (especially the ternary form above that can hide a bit of complexity.

**Review by La0 — CHANGES_REQUESTED** [review]
Side note, the version management could have been a dedicated MR

File: laia/decoders/ctc_greedy_decoder.py (hunk 3 above)
Comment (starride-teklia) [reply laia/decoders/ctc_greedy_decoder.py]: I added some new lines and comments. I also simplified the definition of `zero_tensor` by using `device`.

**Pull request merged**